src/Symfony/Component/Messenger/Bridge/Redis/Transport/Connection.php

```diff
@@ -329,13 +325,13 @@ private function claimOldPendingMessages()
                     ['JUSTID']
                 );
 
-                $this->couldHavePendingMessages = true;
+                $couldHavePendingMessages = $couldHavePendingMessages || $claimResult || $try < 3 && $this->claimOldPendingMessages(++$try);
```

Comment:
I added the `$couldHavePendingMessages || ` part, to account for the variable possibly being set to true on L311, does that make sense?